lib/avro_turf.rb

```diff
@@ -109,8 +109,12 @@ def decode(encoded_data, schema_name: nil, namespace: @namespace)
   def decode_stream(stream, schema_name: nil, namespace: @namespace)
     schema = schema_name && @schema_store.find(schema_name, namespace)
     reader = Avro::IO::DatumReader.new(nil, schema)
-    dr = Avro::DataFile::Reader.new(stream, reader)
-    dr.first
+    entries = Avro::DataFile::Reader.new(stream, reader).entries
+
+    case entries.size
+    when 1 then entries.first
+    when 2.. then entries
+    end
```

Comment:
I would rather return an error here in `else` scenario, but not sure, and wasn't able to TDD it.

Reply:
I don't think we should be changing the behavior of the existing method; rather, we should add a `decode_all_from_stream` method that just returns the `Reader`.